addons/account/models/account_journal.py

```diff
@@ -560,7 +560,7 @@ def write(self, vals):
             if 'restrict_mode_hash_table' in vals and not vals.get('restrict_mode_hash_table'):
                 journal_entry = self.env['account.move'].search([('journal_id', '=', self.id), ('state', '=', 'posted'), ('secure_sequence_number', '!=', 0)], limit=1)
                 if len(journal_entry) > 0:
-                    field_string = self._fields['restrict_mode_hash_table'].get_description(self.env)['string']
+                    field_string = self.fields_get(allfields=['restrict_mode_hash_table'], attributes=['string'])['restrict_mode_hash_table']['string']
```

Comment:
Given the awful parameter name `allfields`, I wouldn't pass it as a named parameter.  When you call `fields_get()`, you know the first parameter is a list of field names
```suggestion
                    field_string = self.fields_get(['restrict_mode_hash_table'], attributes=['string'])['restrict_mode_hash_table']['string']
```